orchagent/main.cpp

```diff
@@ -401,33 +405,44 @@ int main(int argc, char **argv)
         attrs.push_back(attr);
     }
 
-    //Get info required for VOQ system
+    // Instantiate database connectors
+    DBConnector appl_db("APPL_DB", 0);
+    DBConnector config_db("CONFIG_DB", 0);
+    DBConnector state_db("STATE_DB", 0);
+
+    // Get Switch type
+    getCfgSwitchType(&config_db, gMySwitchType);
```

Comment:
Wonderful change. Thanks for working on it.

Could you move these lines up a little bit higher? Should be before 

    if (gMacAddress)
    {
        attr.id = SAI_SWITCH_ATTR_SRC_MAC_ADDRESS;
        memcpy(attr.value.mac, gMacAddress.getMac(), 6);
        attrs.push_back(attr);
    }

It's because not all asics have mac addr, or need to set mac addr. Your gMySwitchType would help to avoid "getting unnecessary mac addr".

Reply:
Yes. Thanks. Fixed as suggested.